### PR TITLE
[PB-665]: feat/user can request access to a file or folder

### PR DESCRIPTION
**Pull request by rafijv**
Added new view that allows users to request access to files or folders

### CODE REVIEW

File: src/app/drive/views/RequestAccess/RequestAccess.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+function RequestAccess(): JSX.Element {
+  const user = useSelector<RootState, UserSettings | undefined>((state) => state.user.user);
+  const { translate } = useTranslationContext();
+
+  const [messageText, setMessageText] = useState<string>('');
+  const [messageTextLimit, setMessageTextLimit] = useState<boolean>(false);
+  const [requestSent, setRequestSent] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (messageText.length >= 950) {
```
Comment (CandelR) [inline src/app/drive/views/RequestAccess/RequestAccess.tsx]: Avoid use magic numbers. Add it to a const in order to know what represents the number or what is its purpose with the name

File: src/app/core/config/app.json [hunk 2]
```diff
@@ -218,6 +218,13 @@
       "path": "/verify-email/:token",
       "exact": false
     },
+    {
+      "id": "request-access",
+      "layout": "empty",
+      "path": "/request-access",
```
Comment (CandelR) [inline src/app/core/config/app.json:224]: Does the url not have to have a file or folder id?

File: src/app/core/config/views.ts [hunk 3]
```diff
@@ -68,6 +69,7 @@ const views: Array<{
   { id: AppView.SharePhotosToken, component: SharePhotosView },
   { id: AppView.RedirectToApp, component: RedirectToAppView },
   { id: AppView.VerifyEmail, component: VerifyEmailView },
+  { id: AppView.RequestAccess, component: RequestAccess },
```
Comment (CandelR) [inline src/app/core/config/views.ts]: Now it is accessible, hide it until the task is completed if this is just the UI.

**Review by CandelR — CHANGES_REQUESTED** [review]
The Change button will be moved out of the div if the email is long.
image

File: src/app/core/config/app.json (hunk 2 above)
Comment (rafijv) [reply src/app/core/config/app.json:224]: I don't know how this is going to work, we will have to take it up again when the back is finished and we start with the logic.

File: src/app/core/config/views.ts (hunk 3 above)
Comment (rafijv) [reply src/app/core/config/views.ts]: I have commented it so it is not available

File: src/app/drive/views/RequestAccess/RequestAccess.tsx (hunk 1 above)
Comment (rafijv) [reply src/app/drive/views/RequestAccess/RequestAccess.tsx]: Done

**Review by CandelR — APPROVED** [review]

**Pull request merged**